rfcs/text/0014_scheduled_reports_new_plugin.md

```diff
@@ -0,0 +1,164 @@
+  open: (urlOrObjectID: string) => void;
+
+  // Simply close the report generation panel.
+  close: () => void;
+}
+```
+
+## 4. Render your page properly
+
+In cases where a report is a rasterized asset (PDF or PNG), your page should take care of when and how this happens. For instance, if your page is located at:
+
+```
+https://www.kibana-is-cool.com/my-dope-kibana-page-a-rama
+```
+
+Then, setting a URL-based argument might make sense in your hook:
+
+```
+https://www.kibana-is-cool.com/my-dope-kibana-page-a-rama?pdf=true
+```
+
+Having something to distinguish whether a page is being exported, or simply rendered as usual, is critical and allows page owners to make conditional decisions based upon that predicate. For instance, app owners can (and should) disable "first-time" wizards, telemetry, and anything else expecting user interactions (modals/prompts/buttons/etc). The browser that reporting uses will appear as a "first-time" user with no cache or cookies set.
+
+## 5. ... Wait
+
+After a report has been submitted, either via a user-initiated action or a programmatic one (not covered in this RFC), reporting will function like it generally has. Changes here are purely internal to the team, and don't require commentary since the implementation isn't of concern to consumers. Once complete, a toast-notification is trigger with a link to download the report (in the case of immediate intervals, or simple no-connector) and a link to the Reporting page. For scheduled reports, a simple success notification is generated at the start to give notice to the user that the report will begin at the requested interval.
+
+## 6. Downloads and exploration of prior reports
+
+Aside from the creation aspect, reporting will continue to own the reading, updating, and deleting of prior-run reports. These actions will continue to reside in our section of the Management are, and future RFCs/development will be required for performing these operations outside of reportings administration page.
+
+# Drawbacks
+
+While migrating reporting to be plugin-oriented presents a good way to delineate ownership, there's a few downsides as well.
+
+- Re-architect how reporting works so that it can operate as plugin versus a standalone plugin.
+- Migration of code into the relevant apps/plugins.
+- No tangible features or performance improvements: this is purely a refactor internally.
+- Documenting and evangelizing of this new procedure.
+- Potential breaking of 3rd party plugins once we extract page-level behaviors.
+
+This list is in combination with all of the other known issues regarding rewrite-or-refactoring.
+
+# Alternatives
+
+The only alternative that comes to mind is integrating future pages and features into our plugin's codebase. This gets us further in the short-term, but presents dramatic costs longer-term as a higher headcount and staffing will be needed to ensure reporting continues to operate normally plus add new features.
```

Comment:
alternative to ? if we are just talking about ownership of code i think there are quite a few. For example, what if apps (that support reporting):
- take care of page rendering (as mentioned above)
- but also take care of showing the generate report button. i think this way we can get rid of one registry.

Reply:
++ for not adding a new registry